src/guide/security/generating-cryptographic-keys.md

```diff
@@ -0,0 +1,225 @@
+# Generating Cryptographic Keys
+
+In the realm of blockchain technology, cryptographic keys play a crucial role in upholding the security and authenticity of data transactions. With Iroha 2, you can create these vital keys for safeguarding your digital assets and communications.
+
+This section describes how to generate keys using `kagami`, a supporting tool shipped alongside Iroha 2.
+
+In the future, alternative methods of generating public keys shall be added.
+
+## Generating Cryptographic Keys with Kagami {#kagami}
+
+::: tip Note
+
+Since `kagami` does not come with a manual page, you can use the `--help` (`-h`) command to retrieve a brief summary of all the usable `kagami` parameters within the CLI you are using.
+
+:::
+
+To generate a new key pair having only just cloned the Iroha repo, run the following command from the project's `root` directory:
+
+```bash
+$ cargo run --bin kagami crypto
```

Comment:
```suggestion
$ cargo run --bin kagami --release -- crypto
```

Or suggest installing the binary system-wide with `cargo`:

```suggestion
$ cargo install --path ./tools/kagami
$ kagami crypto
```

This way in future sections you can just refer to `kagami` instead of `cargo run --bin kagami --release -- ...`